lib/runner/index.js

```diff
@@ -101,35 +104,39 @@ _.assign(Runner, {
      * @private
      *
      * @param {?Collection} collection
-     * @param {?String} groupId
+     * @param {?String} entrypoint folder/request id or name to begin extraction
      * @param {Function} callback
      * @returns {Object}
      * @todo Add support for accepting groupId as an array of ids/names
      */
-    extractRunnableItems: function (collection, groupId, callback) {
-        var idMatch,
-            nameMatch,
-            items = [],
-            group = !groupId && collection; // if a group id is unspecified, the collection is our group
-
-        // if group id is not provided, we try and check if we can find it in the collection
-        if (!group) {
-            collection && collection.forEachItemGroup && collection.forEachItemGroup(function (itemGroup) {
-                // Matching has been done like this to avoid having to go over the collection item groups twice
-                (itemGroup.id === groupId) && (idMatch = itemGroup);
-                (itemGroup.name === groupId) && (nameMatch = itemGroup);
-            });
-
-            group = idMatch || nameMatch; // Prioritise id based matches over name based ones.
+    extractRunnableItems: function (collection, entrypoint, callback) {
+        var items = [],
+            entryNode;
+
+        // if an entrypoint is specified, find the item/item group
+        // looks for first node with matching id or name
+        if (entrypoint) {
+            // @todo: fix this. the check by id should all finish before doing check by name
```

Comment:
Left the original comment here.

IMHO, we do not support this yet and I think we need hold off on this, now that the search goes through requests as well. Collections with names colliding with ids should be very rare and should not penalize performance of normal collections.

If we really want to support it, we have to make it a more performant search. It is common to have collections with thousands of requests.

Reply:
@kunagpal

Reply:
In that case, `.oneDeep` will have to be adjusted to prioritize id matches over name matches.

Reply:
`.oneDeep` prioritizes `id` now see - https://github.com/postmanlabs/postman-collection/blob/develop/lib/collection/item-group.js#L210